pkg/doc/verifiable/credential.go

```diff
@@ -239,48 +244,89 @@ func WithDisabledCustomSchemaCheck() CredentialOpt {
 	}
 }
 
-// NewCredential creates an instance of Verifiable Credential by reading a JSON document from bytes
-func NewCredential(data []byte, opts ...CredentialOpt) (*Credential, error) {
+// WithCredentialBuilder defines custom credential builder which usually used when data model should be extended.
+func WithCredentialBuilder(cb CredentialBuilder) CredentialOpt {
+	return func(opts *credentialOpts) {
+		opts.cb = cb
+	}
+}
+
+func issuerMapping(data []byte, credential *Credential) (err error) {
+	issuerID, issuerName, err := issuerFromBytes(data)
+	if err != nil {
+		err = fmt.Errorf("JSON unmarshalling of Verifiable Credential bytes failed: %w", err)
+		return
```

Comment:
Better to have a explicit return than naked return -  [Named Result Parameters](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters)

Reply:
I opened #216

Reply:
Useful link regarding code review comments! Thanks, I will update this.